test/Simple.cpp

```diff
@@ -251,17 +244,101 @@ int basicTest(int argc, char** argv, const char* protocolVersion, Marshaller<K>
+            std::string script_name("script.js");
+            cacheManager.getCache<std::string,std::string>("___script_cache",false).put(JBasicMarshaller<std::string>::addPreamble(script_name), JBasicMarshaller<std::string>::addPreamble(script));
+            char* execResult = cache.execute(script_name,s);
+            if (!(execResult[3]=='a' && execResult[4]=='b' && execResult[5]=='c'))
+            {
+                std::cerr << "fail: cache.exec() returned unexpected result"<< std::endl;
+                return 1;
+            }
+            delete(execResult);
+        } catch (const Exception& e) {
+            std::cout << "is: " << typeid(e).name() << '\n';
+            std::cerr << "fail unexpected exception: " << e.what() << std::endl;
+            return 1;
+        }
+        std::cout << "PASS: script execution on server" << std::endl;
+
+        cacheManager.stop();
+
+    }
     if (result!=0)
         return result;
 
     std::cout << "Basic Test with JBasicMarshaller" << std::endl;
-    result=basicTest<std::string,std::string>(argc, argv, Configuration::PROTOCOL_VERSION_12,
-                                                new JBasicMarshaller<std::string>(), &Marshaller<std::string>::destroy,
-                                                new JBasicMarshaller<std::string>(), &Marshaller<std::string>::destroy);
+    {
+        ConfigurationBuilder builder;
+        builder.addServer().host(argc > 1 ? argv[1] : "127.0.0.1").port(argc > 2 ? atoi(argv[2]) : 11222).protocolVersion(Configuration::PROTOCOL_VERSION_20);
+        RemoteCacheManager cacheManager(builder.build(), false);
+        JBasicMarshaller<std::string> *km = new JBasicMarshaller<std::string>();
+        JBasicMarshaller<std::string> *vm = new JBasicMarshaller<std::string>();
+        RemoteCache<std::string, std::string> cache = cacheManager.getCache<std::string, std::string>(km,
+                &Marshaller<std::string>::destroy,
+                vm,
+                &Marshaller<std::string>::destroy);
+        cacheManager.start();
+        result = basicTest<std::string, std::string>(cacheManager, cache);
+        try
+        {
+            std::map<std::string,std::string> s;
+            std::string argName = std::string("a");
+            std::string argValue = std::string("b");
+            s.insert(std::pair<std::string, std::string>(argName,argValue));
+            std::string script ("// mode=local,language=javascript\n "
+            "var cache = cacheManager.getCache();\n"
+            "cache.put(\"a\", \"abc\");\n"
+            "cache.put(\"b\", \"b\");\n"
+            "cache.get(\"a\");\n");
+            std::string script_name("script.js");
+            cacheManager.getCache<std::string,std::string>(km,
+                    &Marshaller<std::string>::destroy,
+                    vm,
+                    &Marshaller<std::string>::destroy,"___script_cache").put(script_name, script);
+            char* execResult = cache.execute(script_name,s);
```

Comment:
I guess the result is in JBossMarshalled format. It would be nice to unmarshall it instead of inspecting it as below.